labs/a8.md

```diff
@@ -0,0 +1,745 @@
+obvious: the network allows one to interact with a machine without needing any 
+physical presence. Anyone on the same network as your machine can connect, and 
+therefore attack, your machine. On the internet, this means the whole world can 
+attack your machine. This introduces a number of security considerations when 
+running networked applications.
+
+The first step in securing machines connected to a network, or the open 
+internet, is to make it as hard as possible for attackers to get to your 
+machine. As you may recall from the networking lecture, processes interact with 
+one another over a network by means of sockets and ports. Many protocols listen 
+on well-known ports, such as SSH on port 22, HTTP on port 80, FTP on port 21, 
+etc. We use firewalls and similar software to prevent unauthorized users from 
+connecting to your machine. For example, the student VMs are heavily firewalled 
+- the only traffic allowed from the internet at large is SSH (TCP to port 22), 
+HTTP/S (TCP to port 80, 443, 8000, 8080) and a few other services necessary for 
+labs (e.g. UDP to port 53 for lab 7).
+
+The second step in maintaining network security is to audit which programs are 
+running on your server, and to design your network to reduce the attack vectors 
+on any single machine by separating concerns as much as reasonably possible. 
+For example, it may not be a good idea to run a critical database server and a 
+potentially insecure webserver on the same machine, as a vulnerability in one 
+could easily lead to the compromising of the other. 
+
+Finally, the hardest part of online security is auditing your own code. 
+Ultimately, most security vulnerabilities arise from code you need to run to 
+affect your business functions. For example, the popular CMS software 
+WordPress, which is estimated to power nearly a quarter of all websites, has
+historically been extremely vulnerable to security bugs, especially because people
+like to install WordPress plugins, often written by amateur coders, which are even 
+less secure. Unfortunately, the only real defense against one's own imperfection
+as a programmer is to be vigilant about monitoring one's programs for vulnerabilities
+and servers for intrusion, and following good defensive programming practices on the
+road towards the impossible goal of secure, bug-free programs.
+
+#### Task 7 - Network Security Lab Activity
+
+1. Make a list of all the services running on your student VM that are 
+accessible from the public internet, what user they are running as, and what 
+port they are listening on. Use the tools you learned about in lab 5, such as
+`netstat`, as well as tools like `ps`. You may also want to point `nmap -A`
+at your VM from another machine, such as `tsunami`.
+
+2. Use `less` and `grep` to open up and search your SSH login log file, located 
+in `/var/log/auth.log`. Besides yourself, is anyone trying to log in? Who and 
+why?
```

Comment:
Nice, I like this question especially, I think this'll be quite interesting for students to look at.